### PR TITLE
xeval/evaluator:implement json_unquote

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi, 
     This PR implement function `json_unquote` for evaluator, the related PR in `tidb` is https://github.com/pingcap/tidb/pull/3492,
and the related source files in `tidb` are

- https://github.com/pingcap/tidb/blob/master/expression/builtin_json.go
- https://github.com/pingcap/tidb/blob/master/distsql/xeval/eval_json.go
- https://github.com/pingcap/tidb/blob/master/distsql/xeval/eval.go

And the file datum.rs is the same with https://github.com/pingcap/tikv/pull/1996. we should do review after it has been merged.

- [x] https://github.com/pingcap/tikv/pull/1996

@hhkbp2 @hicqu @lishihai9017 @andelf PTAL

### CODE REVIEW

**Comment by hhkbp2** [comment]
LGTM

File: src/util/codec/datum.rs [hunk 1]
```diff
@@ -273,6 +273,7 @@ impl Datum {
                 Json::String(String::from(data)).cmp(json)
             }
             _ => {
+
```
Comment (hicqu) [inline src/util/codec/datum.rs]: remove this line.

File: src/util/codec/datum.rs [hunk 2]
```diff
@@ -299,6 +300,7 @@ impl Datum {
         Ok(b)
     }
 
+
```
Comment (hicqu) [inline src/util/codec/datum.rs]: remove this line.

**Comment by hicqu** [comment]
LGTM.

File: src/util/xeval/evaluator.rs [hunk 3]
```diff
@@ -225,6 +226,17 @@ impl Evaluator {
...
 
+    fn eval_one_child(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let children = expr.get_children();
+        if children.len() != 1 {
+            return Err(Error::Expr(format!("{:?} need 1 operands but got {}",
+                                           expr.get_tp(),
+                                           children.len())));
+        }
+        let child = try!(self.eval(ctx, &children[0]));
+        Ok(child)
```
Comment (andelf) [inline src/util/xeval/evaluator.rs]: Is self.eval(...) ok ?
Avoid try! And rewraps as Result

**Pull request merged**